src/main/java/com/ibm/cloud/sdk/core/service/exception/ServiceResponseException.java

```diff
@@ -22,23 +30,46 @@
   /** The Constant serialVersionUID. */
   private static final long serialVersionUID = 1L;
 
+  /** Potential error message keys. */
+  private static final String MESSAGE_ERROR = "error";
+  private static final String MESSAGE_ERROR_2 = "error_message";
+  private static final String MESSAGE_ERROR_3 = "message";
+
+  private static final Type debuggingInfoType = new TypeToken<Map<String, Object>>() { }.getType();
+
   /** The status code. */
   private final int statusCode;
 
-  /** The HTTP response. */
-  private final Response response;
+  private String message;
+  private Headers headers;
+  private Map<String, Object> debuggingInfo;
 
   /**
    * Instantiates a new Service Response Exception.
    *
    * @param statusCode the status code
-   * @param message the error message
    * @param response the HTTP response
    */
-  public ServiceResponseException(int statusCode, String message, Response response) {
-    super(message);
+  public ServiceResponseException(int statusCode, Response response) {
+    super();
     this.statusCode = statusCode;
-    this.response = response;
+    this.headers = new Headers(response.headers());
+
+    String responseString = ResponseUtils.getString(response);
+    try {
+      final JsonObject jsonObject = ResponseUtils.getJsonObject(responseString);
+      if (jsonObject.has(MESSAGE_ERROR)) {
```

Comment:
It might be cleaner if the various MESSAGE_ERROR* constants were just a String[] where the elements are an ordered list of the property names that we'll look for.   Then the various if/elseif/.../else blocks would become a loop over the elements in the array and we just bail out after finding the first match.   This way, when (and I mean when) the list of properties changes, it will be easy to change them :)

Reply:
That does sound cleaner. I'll make that change along with (maybe) the one to handle the new error spec.